CSNotificationView/CSNotificationView.m

```diff
@@ -383,6 +383,10 @@ - (void)dismissWithStyle:(CSNotificationViewStyle)style message:(NSString *)mess
 - (CGFloat)topLayoutGuideLengthCalculation
 {
     CGFloat top = MIN([UIApplication sharedApplication].statusBarFrame.size.height, [UIApplication sharedApplication].statusBarFrame.size.width);
+
+    UIView *view = self.parentNavigationController.view ?: self.parentViewController.view;
+    CGPoint origin = [view.superview convertPoint:view.frame.origin toView:[UIApplication sharedApplication].keyWindow.rootViewController.view];
```

Comment:
> when the view controller frame actually has an intersection with the status bar frame

This may be a correlation with your usecase but isn't what we want to express. 
I am unhappy with the situation, too, but I don't like the approach of checking the view's location on screen.

I just spent an hour looking for a solution. Three ways:

* Using the layout guide may work if we hooked into the `parentViewController`'s viewDidLayoutSubviews method. However, this is very invasive behavior. But this way would allow binding the topLayoutGuide's length to the CSNotificationView's height. It remains to be seen whether this is backwards compatible to iOS 7.
* Use the `topLayoutGuide` and deal with the following problem: when transitioning between view controllers, CSNotificationView recalculates its height since a presented view controller may have a navigation bar hidden / present. We do this in `navigationControllerWillShowViewControllerNotification`. If we use topLayoutGuide here, it is not set yet because AutoLayout has not done its work at this point. One could postpone the resize to `navigationControllerDidShowViewControllerNotification` (resulting in a delayed resize animation). Since users already rely on a smooth animation, I don't want to break this.
* Making the height (or `addStatusBarHeight`) a property of CSNotificationView would allow app developers to implement those bindings externally. This may be the easiest solution.

I currently do not have the time to do this myself. If you provide a clean implementation of the `addStatusBarHeight` property that does not break the current behavior, I am happy to merge :)